modules/js/test/test_objdetect.js

```diff
@@ -159,3 +159,73 @@ QUnit.test('Cascade classification', function(assert) {
         locations.delete();
     }
 });
+QUnit.test('QR code detect and decode', function (assert) {
+    // QrCodeDetector
+    {
+        assert.expect(12);
+        let done = assert.async(2);
+
+        const detector = new cv.QRCodeDetector();
+        const imgSingleQrCode = new Image();
+        imgSingleQrCode.crossOrigin = 'anonymous';
+        imgSingleQrCode.onload = function () {
+            const canvas = document.createElement('canvas');
+            const ctx = canvas.getContext('2d');
```

Comment:
Does it work in Node.JS ?

Reply:
You are right, this is not working in Node.js.
In order to provide tests i have seen other JavaScript tests, where images are created from a hardcoded array.
In this test case it would probably be a huge array to create an image containing a QR-code.
Is there a way of loading images, which works both in the browser and in node.js? Otherwise i have to implement two different ways of loading the images.

Reply:
Until we have good images (testdata) support in JS tests infrastructure, lets check the "bindings" existence only (without any attempt to perform accuracy checks).

Please use Mat.zeros(...) as input in tests for now. And check that there are no exceptions raised. Minimal smoke test.

For accuracy demo it would be nice to add tutorial page ([examples](https://docs.opencv.org/3.4/dc/d73/tutorial_js_table_of_contents_objdetect.html)).

Reply:
I have now added tests that do not require any test images.
Therefore I have removed the copying of the test images again.
I will start writing a tutorial page. Should the tutorial include the QR code recognition algorithm? As far as I could understand `QrCodeDetector::detect` uses the algorithm described in this [article](https://www.atlantis-press.com/proceedings/emeit-12/3264).